# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/ShardhaBatra/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### Freestyle – Explainable Future Content Decline Prediction

I am choosing the freestyle option because I want to study whether historical search performance can help predict which content pages may decline in the future. My goal is not only to predict future decline but also to explain the reasons behind each prediction so that content reviewers can understand why a page was recommended. I chose this direction because it combines machine learning with practical decision support and will help me learn feature engineering, time-based validation, and explainable machine learning.



## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### Research Question

Can historical search performance predict which content pages are at risk of future decline, and can those predictions be explained to help reviewers prioritize which pages to inspect first?

### Decision

Help content reviewers decide which pages should be reviewed first based on their predicted risk of future decline.

### Unit of Analysis

One content page.

### Output

A ranked list of content pages with a predicted decline risk score, explanation of the prediction, and review priority.

### Action

Content reviewers can inspect the highest-risk pages first and decide whether they should be refreshed, updated, monitored, or investigated further.

### Cost of a Wrong Recommendation

If a high-risk page is not recommended, it may continue losing traffic and visibility before anyone notices. If a low-risk page is recommended instead, reviewers may waste valuable time reviewing pages that did not need immediate attention.

### Why Data or Machine Learning Can Help

Content performance depends on many signals such as impressions, clicks, CTR, sessions, content age, and search trends. Machine learning can analyze these signals together and help prioritize pages for review more efficiently than manual inspection.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))
print("Pages with Declining Trend:", (df["trend_direction"] == "down").sum())
print("Average CTR:", round(df["ctr"].mean(), 4))

Total Pages: 30000
Pages with Declining Trend: 16262
Average CTR: 0.5107


The starter dataset contains **30,000** content pages. Among them, **16,262** pages show a declining trend, which indicates that there are many pages that may require attention. The average CTR is **0.5107**, showing that click-through performance varies across pages. These numbers suggest that predicting future content decline and prioritizing pages for review is a meaningful machine learning problem to explore over the next seven weeks.


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is a decision-support system, not an automated decision maker. It will use historical search performance to identify pages that may be at risk of future decline and explain the reasons behind each prediction. The results will be observational and predictive rather than causal. This project cannot predict Google's ranking algorithm or guarantee that updating a page will improve its performance. Final decisions should always be made by human reviewers.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.